src/recipes/cache.rs

```diff
@@ -1,20 +1,25 @@
+//! Caching mechanisms.
 use std::thread;
 use std::sync::{Arc, Mutex};
 use std::sync::mpsc;
 use std::sync::mpsc::Sender;
 use std::collections::HashMap;
 use consts::{WatchedEventType, ZkError, ZkState};
+use data::Stat;
 use paths::make_path;
-use proto::WatchedEvent;
+use watch::WatchedEvent;
 use zookeeper::{ZkResult, ZooKeeper};
 use zookeeper_ext::ZooKeeperExt;
 use listeners::{ListenerSet, Subscription};
 
-pub type ChildData = Arc<Vec<u8>>;
+/// Data contents of a znode and associated `Stat`.
+pub type ChildData = Arc<(Vec<u8>, Stat)>;
```

Comment:
Using a cache without the `Stat::version` is dangerous, so I added it. As a side-note, this makes things look more like the standard `ZooKeeper::get_data` calls.